pandas/tests/indexing/test_loc.py

```diff
@@ -1106,3 +1107,137 @@ def test_loc_with_period_index_indexer():
     tm.assert_frame_equal(df, df.loc[list(idx)])
     tm.assert_frame_equal(df.iloc[0:5], df.loc[idx[0:5]])
     tm.assert_frame_equal(df, df.loc[list(idx)])
+
+
+def test_loc_setitem_df_datetime64tz_full_column_with_index():
+    df = pd.DataFrame(
+        pd.date_range("2020-01-01", "2020-01-06", 6, tz=timezone.utc), columns=["data"]
+    )
+    df2 = pd.DataFrame(index=df.index)
+    df2.loc[df.index, "data"] = df["data"]
+    tm.assert_frame_equal(df, df2)
+
+
+def test_loc_setitem_df_datetime64tz_slice():
+    df = pd.DataFrame(
+        pd.date_range("2020-01-01", "2020-01-06", 6, tz=timezone.utc), columns=["data"]
+    )
+    df2 = pd.DataFrame(index=df.index)
+    expected = pd.DataFrame(
+        [
+            pd.Timestamp("2020-01-01", tz=timezone.utc),
+            pd.Timestamp("2020-01-02", tz=timezone.utc),
+            pd.Timestamp("2020-01-03", tz=timezone.utc),
+            pd.NaT,
+            pd.NaT,
+            pd.NaT,
+        ],
+        columns=["data"],
+        dtype="object",
+    )
+    df2.loc[df.index[:3], "data"] = df["data"][:3]
+    tm.assert_frame_equal(df2, expected)
+
+
+def test_loc_setitem_df_datetime64tz_column_without_index():
```

Comment:
you don't need to repeat things like this, instead we aleady have generic tests in : pandas/tests/extension//base/setitem.py

Reply:
@h-vishal this is the key comment. Please do not repeat things like this. We already have fixtures for all of these.

Reply:
@jreback I have pushed a commit with the requested changes.

Reply:
it doesn't look like anything has changed. there should be NO tests here, rather in pandas/tests/extension/base/setitem